Custom Next.js server throws 'Could not find ${fileToRun}'

### Current Behavior

The previous issue (#17946) was meant to be fixed in #17884, which was released in [16.5.0](https://github.com/nrwl/nx/releases/tag/16.5.0).

However, this issue remains in the latest version of Nx (16.5.5).

### Expected Behavior

The custom Next.js server should work.

### GitHub Repo

_No response_

### Steps to Reproduce

1. Create a new workspace with a Next.js application.
2. Add a custom server via `nx g custom-server [app-name]`
3. Attempt to serve the app via `nx serve [app-name]`

### Nx Report

```shell
>  NX   Report complete - copy this into the issue template

   Node   : 16.18.0
   OS     : darwin-arm64
   npm    : 8.19.2
   
   nx                 : 16.5.5
   @nx/js             : 16.5.5
   @nx/jest           : 16.5.5
   @nx/linter         : 16.5.5
   @nx/workspace      : 16.5.5
   @nx/cypress        : 16.5.5
   @nx/devkit         : 16.5.5
   @nx/eslint-plugin  : 16.5.5
   @nx/next           : 16.5.5
   @nx/react          : 16.5.5
   @nrwl/tao          : 16.5.5
   @nx/web            : 16.5.5
   typescript         : 5.1.6
```


### Failure Logs

_No response_

### Operating System

- [X] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

_No response_

One pre-requisite here is the custom-server requires a production build located inside `dist/`

However, I think the problem here is that our generator for custom-server sets the default mode to development.
Which is why you might get an error:
- Error: Cannot find 'pages/' or `'app/'`
- Error: Cannot find module '../src/utils/config'

It should be set to production as we are using a production build (located in `dist/`) to serve your Next.js app via custom-server. 

If you want/need to run in dev mode you need to run the default `serve` target or at least with the `@nx/next:server` executor

--- a/packages/next/plugins/with-nx.ts
+++ b/packages/next/plugins/with-nx.ts
@@ -239,7 +239,7 @@ function withNx(
 
       const userWebpackConfig = nextConfig.webpack;
 
-      const { createWebpackConfig } = require('../src/utils/config');
+      const { createWebpackConfig } = require('@nx/next/src/utils/config');
       nextConfig.webpack = (a, b) =>
         createWebpackConfig(
           workspaceRoot,

--- a/packages/next/src/generators/custom-server/custom-server.ts
+++ b/packages/next/src/generators/custom-server/custom-server.ts
@@ -81,7 +81,7 @@ export async function customServerGenerator(
 
   project.targets['serve-custom-server'] = {
     executor: '@nx/js:node',
-    defaultConfiguration: 'development',
+    defaultConfiguration: 'production',
     options: {
       buildTarget: `${options.project}:build-custom-server`,
     },

